internal/vulnstore/postgres/gc.go

```diff
@@ -197,126 +197,27 @@ WHERE array_length(ordered_ops.refs, 1) > $2;
-		// note arguments via closure
-		err := func() error {
-			start := time.Now()
-
-			rows, err := pool.Query(ctx, paginatedSelect, updater, largestID)
-			if err != nil {
-				return err
-			}
-			defer rows.Close()
-			for rows.Next() {
-				var id int64
-				err := rows.Scan(&id)
-				if err != nil {
-					return fmt.Errorf("encountered error while scanning vuln id: %v", err)
-				}
-				ids = append(ids, id)
-			}
-			if rows.Err() != nil {
-				return rows.Err()
-			}
-
-			gcCounter.WithLabelValues("paginatedSelect").Add(1)
-			gcDuration.WithLabelValues("paginatedSelect").Observe(time.Since(start).Seconds())
-			return nil
-		}()
-		if err != nil {
-			return fmt.Errorf("failed to collect vuln ids for updater %v: %v", updater, err)
-		}
-
-		// if the ids array is empty we processed all
-		// vulns, we terminate the entire function here.
-		if len(ids) == 0 {
-			return nil
-		}
-
-		// keep track of largest id for next paginated request.
-		largestID = ids[len(ids)-1]
-
-		// cross reference uo_vuln table to determine which ids should be
-		// deleted. these operations are small and fast so batching is optimal.
-		// note arguments via closure
-		//
-		// a new batch is returned containing, if any, delete queries to run.
-		toDelete, err := func() (*pgx.Batch, error) {
-			b := &pgx.Batch{}
-			toDelete := &pgx.Batch{}
-
-			for _, id := range ids {
-				b.Queue(shouldDelete, id)
-			}
-
-			start := time.Now()
-			res := pool.SendBatch(ctx, b)
-			defer res.Close()
-
-			var do bool
-			for i := 0; i < b.Len(); i++ {
-				err = res.QueryRow().Scan(&do)
-				if err != nil {
-					return nil, fmt.Errorf("failed while scanning existence check boolean: %v", err)
-				}
-				if do {
-					toDelete.Queue(deleteVuln, ids[i])
-				}
-			}
-
-			gcCounter.WithLabelValues("shoulddelete_batch").Add(1)
-			gcDuration.WithLabelValues("shoulddelete_batch").Observe(time.Since(start).Seconds())
-
-			return toDelete, nil
-		}()
-		if err != nil {
-			return err
-		}
-
-		// perform the delete branch
-		// note arguments via closure
-		err = func() error {
-			if toDelete.Len() == 0 {
-				return nil
-			}
-
-			start := time.Now()
-			res := pool.SendBatch(ctx, toDelete)
-			defer res.Close()
-
-			for i := 0; i < toDelete.Len(); i++ {
-				_, err := res.Exec()
-				if err != nil {
-					return fmt.Errorf("failed while exec'ing vuln delete: %v", err)
-				}
-			}
-			gcCounter.WithLabelValues("deletevuln_batch").Add(1)
-			gcDuration.WithLabelValues("deletevuln_batch").Observe(time.Since(start).Seconds())
-			return nil
-		}()
-		if err != nil {
-			return err
-		}
+	start := time.Now()
+	zlog.Debug(ctx).
+		Str("updater", updater).
+		Msg("Starting clean up")
+	_, err := pool.Exec(ctx, deleteOrphanedVulns, updater)
+	if err != nil {
+		return fmt.Errorf("failed while exec'ing vuln delete: %v", err)
 	}
+	gcCounter.WithLabelValues("deleteVulns").Add(1)
+	gcDuration.WithLabelValues("deleteVulns").Observe(time.Since(start).Seconds())
```

Comment:
```suggestion
	gcCounter.WithLabelValues("deleteOrphanedVulns").Add(1)
	gcDuration.WithLabelValues("deleteOrphanedVulns").Observe(time.Since(start).Seconds())
```

Reply:
Should these counters get tickled before the error is checked? If the query fails, they don't count as written.

Reply:
I think probably not, if there's an error the latency is at best a useless metrics and at worse misleading. The count, should probably include another label that is success/failure (this would also be very useful for update operations).

Reply:
```suggestion
	gcCounter.WithLabelValues("deleteVulns", "true").Inc()
	gcDuration.WithLabelValues("deleteVulns").Observe(time.Since(start).Seconds())
```

Then:
```go
	if err != nil {
		gcCounter.WithLabelValues("deleteVulns", "false").Inc()
		return fmt.Errorf("failed while exec'ing vuln delete: %w", err)
	}
```